### ISSUE TITLE
Option to select and save best model based upon different metric?

### ISSUE THREAD

**Issue by lydiakatsis** [kitzeslab/opensoundscape#345]
Is there a way to choose a different metric from F1 to save and update the best model? E.g. F2 score?

**Comment by sammlapp** [comment]
There isn't. If you wanted to save a different metric, I would recommend training one epoch at a time and performing whatever logic you want to after each epoch. Eg,
```
model.bestF2 = 0
for epoch in num_epochs:
     model.train(... epochs=1)
     F2 = calculate_f2()
     if F2 > model.bestF2:
        model.save(...)
        model.bestF2 = F2
```

**Comment by sammlapp** [comment]
I don't think we need to implement more flexible choices at this time, so I'm going to close this - but you can re-open it if you have further related questions
